app/src/main/java/com/codesoom/assignment/member/application/command/MemberCommandService.java

```diff
@@ -0,0 +1,14 @@
+package com.codesoom.assignment.member.application.command;
+
+import com.codesoom.assignment.member.application.MemberCommand;
+import com.codesoom.assignment.member.application.MemberInfo;
+
+public interface MemberCommandService {
+
+    MemberInfo createMember(MemberCommand.Register command);
+
+    MemberInfo updateMember(MemberCommand.UpdateRequest command);
```

Comment:
`Member`로도 충분하지 않을까요?
```suggestion
    Member createMember(MemberCommand.Register command);

    Member updateMember(MemberCommand.UpdateRequest command);
```

Reply:
Member를 리턴하게 될 경우 엔티티가 그대로 API 스펙이 되어버릴꺼 같은데 괜찮을까요??
API스펙이 바뀌게 되면 엔티티도 변경될거 같아서 Member로 리턴하지않고 MemberInfo로 변환해서 리턴한건데
혹시 컨트롤러쪽에서 리턴할때 MemberInfo로 변환하라는 말씀이실까요??

Reply:
제가 제안하는 건 어디까지나 제안이라는 점을 고려해 주세요. 만약 작업하신 방향이 맞다고 생각하시면 장점과 단점을 잘 비교해서 저를 설득하시면 됩니다.

첫째줄에서 말씀하신 것은 저도 동의합니다. 엔티티가 그대로 HTTP 응답으로 나가게 되면 내부 구현 유출이 되죠.
하지만 여기는 컨트롤러가 아니고 서비스죠. 컨트롤러를 안티 커럽션 레이어로 생각한다면 그 이전까지는 엔티티를 자유롭게 사용하고, 컨트롤러 선에서 내보낼 때에는 응답에 특화된 타입을 사용하는 것을 고려할 수 있겠습니다.

Reply:
한편으로는 서비스가 꼭 HTTP 컨트롤러하고만 연결되지 않을 수 있다는 것도 고려하셔야 합니다.